pulsar-broker/src/main/java/org/apache/pulsar/broker/service/Topic.java

```diff
@@ -161,8 +161,17 @@ default boolean isChunked() {
      * @param option
      * @return
      */
+    @Deprecated
     CompletableFuture<Consumer> subscribe(SubscriptionOption option);
 
+    /**
+     * Subscribe a topic.
+     *
+     * @param autoShrink whether ConcurrentLongLongPairHashMap supports automatic shrinking.
+     * @return
+     */
+    CompletableFuture<Consumer> subscribe(SubscriptionOption option, boolean autoShrink);
```

Comment:
I believe that adding a new method for "autoShrink" is not necessary.
also the variable name should be more meaningful.

Reply:
Do you mean adding the autoShrink parameter directly on the original method?

Reply:
this is global parameter, maybe you have another way to pass this way.
every Consumer will use the same value.

if you add a "parameter" in the function it means that it would be possible to see different values and this is not true

I believe that overall we don't need a configuration flag at all.

your change is good and you demonstrated that it is needed 

cc @merlimat

Reply:
> this is global parameter, maybe you have another way to pass this way.
every Consumer will use the same value.

if you add a "parameter" in the function it means that it would be possible to see different values and this is not true

OK  , I will fix